lib/ansible/modules/cloud/amazon/aws_ses_identity_policy.py

```diff
@@ -0,0 +1,199 @@
+    # In principle we should be able to just return the response, but given
+    # eventual consistency behaviours in AWS it's plausible that we could
+    # end up with a list that doesn't contain the policy we just added.
+    # So out of paranoia check for this case and if we're missing the policy
+    # just make sure it's present.
+    #
+    # As a nice side benefit this also means the return is correct in check mode
+    try:
+        policies_present = connection.list_identity_policies(Identity=identity)['PolicyNames']
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to list identity policies')
+    if policy_name is not None and policy_name not in policies_present:
+        policies_present = list(policies_present)
+        policies_present.append(policy_name)
+    module.exit_json(
+        changed=changed,
+        policies=policies_present,
+    )
+
+
+def delete_identity_policy(connection, module):
+    identity = module.params.get('identity')
+    policy_name = module.params.get('policy_name')
+
+    changed = False
+    try:
+        policies_present = connection.list_identity_policies(Identity=identity)['PolicyNames']
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg='Failed to list identity policies')
+    if policy_name in policies_present:
+        try:
+            if not module.check_mode:
+                connection.delete_identity_policy(Identity=identity, PolicyName=policy_name)
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg='Failed to delete identity policy {policy}'.format(policy=policy_name))
+        changed = True
+        policies_present = list(policies_present)
+        policies_present.remove(policy_name)
+
+    module.exit_json(
+        changed=changed,
+        policies=policies_present,
+    )
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'identity': dict(required=True, type='str'),
+            'state': dict(default='present', choices=['present', 'absent']),
+            'policy_name': dict(required=True, type='str'),
+            'policy': dict(type='json', default=None),
+        },
+        required_if=[['state', 'present', ['policy']]],
+        supports_check_mode=True,
+    )
+
+    # Allow up to 10 attempts to call the SES APIs before giving up (9 retries).
+    # SES APIs seem to have a much lower throttling threshold than most of the rest of the AWS APIs.
+    # Docs say 1 call per second. This shouldn't actually be a big problem for normal usage, but
+    # the ansible build runs multiple instances of the test in parallel.
+    # As a result there are build failures due to throttling that exceeds boto's default retries.
+    # The back-off is exponential, so upping the retry attempts allows multiple parallel runs
+    # to succeed.
+    boto_core_config = Config(retries={'max_attempts': 9})
```

Comment:
Is there a reason not to use AWSRetry? jittered_backoff is nice.

Reply:
There is a small benefit of this approach. Since the low throttling limits seem to apply to just about all the SES APIs, doing it with botocore config means it applies across the board. Where as AWSRetry has to be applied at each callsite.

To use as a decorator I'll have to pull out every boto call into a method while still doing the error handling in the calling method. I think this might get sufficiently verbose that it would obscure the function. 

But the jittered backoff does look nice, I'll have another look at how to use AWSRetry at all calls without making it hard to follow the code. 

Would you be open to an extension of AWSRetry that wraps a boto client and automatically decorates all the calls?

I'm not not sure if I can make that work. But if I can it feels like a nice way of handling this case.  Unless you feel that would add confusing complexity. 

If I can't make a general wrapping work, or you'd prefer to avoid that approach,  I'll go through and decorate all the call sites with AWSRetry.